src/lib/frontend/d_cnf.ml

```diff
@@ -918,11 +1040,210 @@ let rec mk_expr ?(loc = Loc.dummy) ?(name_base = "")
               | _ -> assert false
             end
 
+          | B.Bitv_repeat { n; k }, [t] ->
+            mk_bitv_repeat n k (aux_mk_expr t)
+
+          | B.Bitv_zero_extend { n; k }, [t] ->
+            mk_bitv_extend n false k (aux_mk_expr t)
+
+          | B.Bitv_sign_extend { n; k }, [t] ->
+            mk_bitv_extend n true k (aux_mk_expr t)
+
+          | B.Bitv_rotate_left { n; i }, [t] ->
+            let t' = aux_mk_expr t in
+            if n < 2 then t' else mk_rotate_left n i t'
+
+          | B.Bitv_rotate_right { n; i }, [t] ->
+            let t' = aux_mk_expr t in
+            if n < 2 then t' else mk_rotate_right n i t'
+
+          | B.Bitv_neg n, [t] ->
+            mk_bvneg n (aux_mk_expr t)
+
+          | B.Bitv_not _, [{
+              term_descr = App
+                  ({ term_descr = Cst {builtin = B.Bitv_not _; _ }; _ }, _, [t])
+            ; _ }] ->
+            (* TODO: do the simplification in a smarter smarter way
+               should (bvnot (bvconcat(t1, bvnot t2)))
+               be translated to (bvconcat(bvnot t1, t2))? *)
```

Comment:
Is the frontend really the right place to do these simplifications rather than the bitvector theory?

Reply:
Outdated comment, the simplification are indeed done in the theory.